pkg/server/image_status.go

```diff
@@ -17,14 +17,45 @@ limitations under the License.
 package server
 
 import (
-	"errors"
-
 	"golang.org/x/net/context"
 
 	"k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
 )
 
-// ImageStatus returns the status of the image, returns nil if the image doesn't present.
+// ImageStatus returns the status of the image, returns nil if the image isn't present.
 func (c *criContainerdService) ImageStatus(ctx context.Context, r *runtime.ImageStatusRequest) (*runtime.ImageStatusResponse, error) {
-	return nil, errors.New("not implemented")
+	ref := r.GetImage().GetImage()
+	// if the passed ref is a digest.. and is stored the following get should work
```

Comment:
Easy enough to do the get() to see if it's a full sha###: value digest previously used or not. Matching a sha pattern isn't proof though that it isn't some sha256.io/ :-) dunno..

As for the rest TODO added :-)